src/common/Gateway.sol

```diff
@@ -142,11 +135,11 @@ contract Gateway is Auth, Recoverable, IGateway {
     //----------------------------------------------------------------------------------------------
 
     /// @inheritdoc IGateway
-    function send(uint16 centrifugeId, bytes calldata message, uint128 extraGasLimit)
+    function send(uint16 centrifugeId, bytes calldata message, uint128 extraGasLimit, address refund)
```

Comment:
Curious if we should add a safety-check for `refund != address(0)` when `msg.value > 0` to mitigate loss of funds, particularly because there are code paths with `refund == address(0)` that assume no payment due to unpaid mode.

Reply:
Good question. I don't think so because if the user choose address(0) as refunds it means: burns any reminder it is.

Right now, the codepaths that use `address(0)` don't have any payment. But if it would have it, they didn't know where to refund it, so `address(0)` should be the default as way to say "I don't care"